src/sage/groups/free_group.py

```diff
@@ -775,7 +778,7 @@ def __init__(self, generator_names, libgap_free_group=None):
         if not generator_names:
             cat = Groups().Finite()
         else:
-            cat = Groups().Infinite()
+            cat = Groups().Infinite().FinitelyGenerated()
         Group.__init__(self, category=cat)
```

Comment:
Is there no *infinitely-generated* free group in GAP?

Reply:
I found this from the GAP documentation:

> Called in the fourth form, FreeGroup returns a free group on infinitely many generators. The optional argument name must be a string; its default value is "f", and the optional argument init must be a finite list of nonempty strings; its default value is an empty list. The generators are initially labelled according to the list init, followed by namei for each i in the range from Length(init)+1 to infinity.

Source: https://docs.gap-system.org/doc/ref/chap37_mj.html#X8215999E835290F0

Not relevant here?

Reply:
As far as I can tell, our `FreeGroup` class cannot create free groups with infinitely many generators. If fails by trying to make the list of all names. For example, try `FreeGroup(NN)`

Reply:
Yes. It seems that `FreeGroup_class` wraps only finitely-generated free groups of GAP, though GAP itself supports infinitely-generated free groups.